### PR TITLE
WebUI - Improved "dark" theme

### PR DESCRIPTION
**Pull request by facuxt**
TODO: tune rrdtool to display graphs in dark mode.
image

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: app/Http/ViewComposers/MenuComposer.php [hunk 1]
```diff
@@ -62,7 +62,7 @@ public function compose(View $view)
         // end global Settings
 
         //TODO: should be handled via CSS Themes
-        $vars['navbar'] = in_array($site_style, ['mono', 'dark']) ? 'navbar-inverse' : '';
```
Comment (Jellyfrog) [inline app/Http/ViewComposers/MenuComposer.php:65]: if this is no longer needed it should be removed from all templates and so that use it.

Comment (facuxt) [reply app/Http/ViewComposers/MenuComposer.php:65]: agree, i dont know where its used that variable... so i didnt removed it completely.

Comment (Jellyfrog) [reply app/Http/ViewComposers/MenuComposer.php:65]: Remove it and search for {{ $navbar}} to start with.

Comment (SourceDoctor) [reply app/Http/ViewComposers/MenuComposer.php:65]: if it's really not more used, for my intention TODO-Line should be removed also

**Comment by kkrumm1** [comment]
Looks great! needs some tweaks here and there.  Also, "mono" is used as a theme. 

Screen Shot 2020-04-16 at 7 42 42 AM

**Comment by kkrumm1** [comment]
When I apply your changes and use mono it breaks my navbar.
Screen Shot 2020-04-16 at 7 46 11 AM

**Comment by kkrumm1** [comment]
Screen Shot 2020-04-16 at 7 47 10 AM
Nav bar ^^ with mono theme. This needs to stay.

**Comment by kkrumm1** [comment]
@facuxt Anyword on this?

**Comment by facuxt** [comment]
Im really sorry i cannot continue working on this, i have too much work at the moment, i will try to get this done by the weekend.

**Comment by kkrumm1** [comment]
No worries.

**Comment by facuxt** [comment]
What do you think about this background color in the widgets. i think, the next step should be mess with the rddtool to make inverted-color graphs.
image

Does anyone knows how to make rddtool to make inverted color graphs...   any suggestion is valid here.

**Comment by facuxt** [comment]
With the last commit, i fixed the mono theme navbar, and removed the red color from some of the icons, make the background of the gridster darker, and fixed an issue with some links in the tables.
image
image
image

**Comment by svenvg93** [comment]
> With the last commit, i fixed the mono theme navbar, and removed the red color from some of the icons, make the background of the gridster darker, and fixed an issue with some links in the tables.
> image
> image
> image

Looks Awesome! 
Would be nice to have the RDD text in with, because now you cannot read them.

**Comment by facuxt** [comment]
> > With the last commit, i fixed the mono theme navbar, and removed the red color from some of the icons, make the background of the gridster darker, and fixed an issue with some links in the tables.
> > image
> > image
> > image
> 
> Looks Awesome!
> Would be nice to have the RDD text in with, because now you cannot read them.
yes, but i cannot find a way to do that... Maybe some of the masters here knows a way 
@murrant

**Comment by facuxt** [comment]
I found it... i made some changes in the color scheme in the graphs... now the look awesome!
image
image

**Comment by breeman1** [comment]
This is lovely. Just a suggestion from my side; will it not be better to also make the color of the other icons the same as those in the navigation. They are almost invisible in their current black color, see below.
Snap1

**Comment by breeman1** [comment]
Perhaps even color of the font (white) in the sub menu can be made to be same as that of the navigation menu (gray) to improve visual coherence throughout the theme.

**Comment by facuxt** [comment]
> Perhaps even color of the font (white) in the sub menu can be made to be same as that of the navigation menu (gray) to improve visual coherence throughout the theme.

You are right, i've made a commit with your suggested changes and some more..

- Icons and links now are grey
image
- Modernized "tabs" style into pills, 
image
- button pagemenu-selected fixed
image

**Comment by breeman1** [comment]
Looks really cool, thanks.

**Comment by breeman1** [comment]
Just a quick question. How will this theme deal device symbols that are black and white i.e pfSense? I am just concerned they might be invisible. Below is how the pfSense symbol is currently displayed in our production install.
image

**Comment by facuxt** [comment]
> Just a quick question. How will this theme deal device symbols that are black and white i.e pfSense? I am just concerned they might be invisible. Below is how the pfSense symbol is currently displayed in our production install.
> image

i think i could add a css class to add a little white glow around the logos...  what is your opinion about that?

**Comment by breeman1** [comment]
Thanks, that could work I think. Would you mind posting a screenshot of what it could look like?

**Comment by facuxt** [comment]
> Thanks, that could work I think. Would you mind posting a screenshot of what it could look like?

image

like this without touching anything... see the part of the ubiquity logo that its black... i dont think its necessary to add a 1990 glow on it 😋

**Comment by breeman1** [comment]
I think it is because the eye/brain realizes the fact that the black is part of the gray and blue, making the need for the glow unnecessary in this case. Not sure about an all-black logo though, I think it will look strange without the glow. Just just my thought.

That said, I noticed that the logo of the actual pfSense firewall is not black and white, but looks like below;
image
Perhaps LibreNMS should rather adopt this one instead of the black and white one. In that case the problem is solved. However this could still be an issue though for some other logos with mostly black shades in their design.

**Comment by facuxt** [comment]
Update:
Improved table headings.
image

Changed the line color of Ping and Poller
image

**Comment by facuxt** [comment]
> I think it is because the eye/brain realizes the fact that the black is part of the gray and blue, making the need for the glow unnecessary in this case. Not sure about an all-black logo though, I think it will look strange without the glow. Just just my thought.
> 
> That said, I noticed that the logo of the actual pfSense firewall is not black and white, but looks like below;
> image
> Perhaps LibreNMS should rather adopt this one instead of the black and white one. In that case the problem is solved. However this could still be an issue though for some other logos with mostly black shades in their design.

I suggest you to try this modifications and see for yourself if you can see the logo.

File: app/Http/ViewComposers/MenuComposer.php (hunk 1 above)
Comment (facuxt) [reply app/Http/ViewComposers/MenuComposer.php:65]: Already fixed it, its needed by the "mono"theme... so i left that sentence, with "mono" only.

**Comment by facuxt** [comment]
@Jellyfrog @kkrumm1 @PipoCanaja any chances to see it merged in the following milestones? can you tell me if there is anything pending to do on my side?

thanks

**Comment by SourceDoctor** [comment]
@facuxt therefore travis has to run successfull. You've to merge against master to fix the issues which travis has

**Comment by facuxt** [comment]
> @facuxt therefore travis has to run successfull. You've to merge against master to fix the issues which travis has

Thanks, i did the merge from master and uploaded in this branch/pr

**Comment by facuxt** [comment]
> @facuxt therefore travis has to run successfull. You've to merge against master to fix the issues which travis has

Any idea how to get this further down in the development line? 

Thanks

**Comment by kkrumm1** [comment]
We need people to test this out first. @librenms/reviewers

**Comment by facuxt** [comment]
> We need people to test this out first. @librenms/reviewers

Thanks, i 

> We need people to test this out first. @librenms/reviewers

Great, i will follow updates on this PR, if you need me to fix something, ill be around.

**Comment by facuxt** [comment]
@kkrumm1 what happened?

**Comment by kkrumm1** [comment]
Merged it. Look like you were ready.

**Comment by facuxt** [comment]
> Merged it. Look like you were ready.

i mean the "revert" commit.. dont know what happened there.

**Comment by TheGreatDoc** [comment]
@facuxt @kkrumm1 Are you sure this was ok to merge?

This is how looks now my install with dark theme:
image

**Comment by kkrumm1** [comment]
Yes, you need to clear your cache.

> On May 22, 2020, at 11:04 PM, TheGreatDoc <notifications@github.com> wrote:
> 
> ﻿
> @facuxt @kkrumm1 Are you sure this was ok to merge?
> 
> This is how looks now my install with dark theme:
> 
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or unsubscribe.

-- 










CONFIDENTIALITY NOTICE: This email & attached documents may 
contain confidential information. All information is intended only for the 
use of the named recipient. If you are not the named recipient, you are not 
authorized to read, disclose, copy, distribute or take any action in 
reliance on the information and any action other than immediate delivery to 
the named recipient is strictly prohibited. If you have received this email 
in error, do not read the information and please immediately notify sender 
by telephone to arrange for a return of the original documents. If you are 
the named recipient you are not authorized to reveal any of this 
information to any other unauthorized person. If you did not receive all 
pages listed or if pages are not legible, please immediately notify sender 
by phone.

**Comment by TheGreatDoc** [comment]
Seems still needs a little bit more work (or is just me?)
image

And Device->Ports section too:
image

I'll keep testing it a bit and will comment again if I see more places that needs to be updated

**Comment by TheGreatDoc** [comment]
Device -> Alerts (When no alerts)
image

**Comment by kkrumm1** [comment]
@facuxt you think you can submit a PR to fix those pages @TheGreatDoc is talking about?

**Comment by facuxt** [comment]
yes, ofcourse, i will try to replicate those issues and fix them.

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-64-release-changelog-may-2020/12290/1

**Comment by BloodyIron** [comment]
I've switched the Dark Theme on and been loving it for a long while now! However there is one area I really would like to get adjusted, rrd related. Specifically I'm talking about Apps (OS Updates/Proxmox/etc).

When I go into the "OS Updates" section of Apps, only the single day graphs are shown, not multiple graphs per device. But this does look good with the dark theme. So if we could get more graphs on this page for history, that would be great, in similar vein to how most other areas of libreNMS do it.

When I go into the Overview section of Apps, the graphs that I see there have white background with green graph, unlike the black background in the OS Updates section itself.  These graphs in the Overview section show up for OS Updates and Proxmox (I don't have other App stuff setup yet). So this really counters the point of a dark theme for this section.

If we could get this addressed at some point that'd be great. :)

**Pull request merged**